frontend/amundsen_application/static/js/ducks/search/filters/reducer.ts

```diff
@@ -46,10 +47,33 @@ export interface ResourceFilterReducerState {
   };
 }
 
+export function getDefaultFiltersForResource(
+  resourceType: ResourceType
+): ResourceFilterReducerState {
+  const defaultFilters = {};
+  const resourceConfig = appConfig.resourceConfig[resourceType];
+  resourceConfig.filterCategories.forEach((filter) => {
+    const filterDefaultValue: string[] = filter.defaultValue;
+    if (filterDefaultValue && filterDefaultValue !== []) {
+      defaultFilters[filter.categoryId] = {
+        value: filterDefaultValue.join(),
+        filterOperation: filter.allowableOperation,
+      };
+    }
+  });
```

Comment:
Let's try to do this logic with a reduce and a filter instead of this 'forEach'.
It would be something like:
```
defaultFilters = filterCategories.filter(hasDefaultValue).reduce((acc, filter) => {
  return {
       ...acc, 
       [filter.categoryId]: {
           value: filterDefaultValue.join(),
            filterOperation: filter.allowableOperation
        }
}, {});
```